### PR TITLE
UAR-762: Add statement validation error page

### PR DESCRIPTION
**Pull request by conuirchgov**
### JIRA link

https://companieshouse.atlassian.net/browse/UAR-762

### Change description

Added page to display validation errors with beneficial owner statements made by the user

Associated PR for the new `.update-bo-statement` style:
- https://github.com/companieshouse/cdn.ch.gov.uk/pull/574

### Work checklist

- [ ] Tests added where applicable
- [ ] UI changes meet accessibility criteria

### Merge instructions

We are committed to keeping commit history clean, consistent and linear. To achieve this, this commit should be structured as follows:

```
<type>[optional scope]: <description>
```

and contain the following structural elements:

- fix: a commit that patches a bug in your codebase (this correlates with PATCH in semantic versioning),
- feat: a commit that introduces a new feature to the codebase (this correlates with MINOR in semantic versioning),
- BREAKING CHANGE: a commit that has a footer `BREAKING CHANGE:` introduces a breaking API change (correlating with MAJOR in semantic versioning). A BREAKING CHANGE can be part of commits of any type,
- types other than `fix:` and `feat:` are allowed, for example `build:`, `chore:`, `ci:`, `docs:`, `style:`, `refactor:`, `perf:`, `test:`, and others,
- footers other than `BREAKING CHANGE: <description>` may be provided.

### CODE REVIEW

File: src/controllers/update/update.statement.validation.errors.controller.ts [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+    const appData = getApplicationData(req.session);
+    const errors = validationResult(req);
+    const inNoChangeJourney = !!appData.update?.no_change;
+
+    if (!errors.isEmpty()) {
+      return renderPage(res, appData, inNoChangeJourney, formatValidationError(errors.array()));
+    }
+
+    if (req.body.statement_resolution === 'statement-resolution-change-information') {
+      return res.redirect(inNoChangeJourney
```
Comment (AndreKramer7654) [inline src/controllers/update/update.statement.validation.errors.controller.ts:42]: a getChangeUrl(inNoChangeJourney) method to mirror what you did in the else?

File: src/utils/navigation.ts [hunk 2]
```diff
@@ -379,4 +379,9 @@ export const NAVIGATION: Navigation = {
     previousPage: () => config.UPDATE_ANY_TRUSTS_INVOLVED_URL,
     nextPage: [config.UPDATE_ANY_TRUSTS_INVOLVED_URL]
   },
+  [config.UPDATE_STATEMENT_VALIDATION_ERRORS_URL]: {
+    currentPage: config.UPDATE_STATEMENT_VALIDATION_ERRORS_PAGE,
+    previousPage: () => config.UPDATE_ANY_TRUSTS_INVOLVED_URL,
```
Comment (AndreKramer7654) [inline src/utils/navigation.ts:384]: would it not be the statement page/url?

File: src/utils/navigation.ts [hunk 3]
```diff
@@ -379,4 +379,9 @@ export const NAVIGATION: Navigation = {
...
+    nextPage: [config.UPDATE_BENEFICIAL_OWNER_TYPE_URL, config.UPDATE_DO_YOU_WANT_TO_MAKE_OE_CHANGE_URL],
```
Comment (AndreKramer7654) [inline src/utils/navigation.ts:385]: from post method - also UPDATE_NO_CHANGE_BENEFICIAL_OWNER_STATEMENTS_URL
    : UPDATE_BENEFICIAL_OWNER_STATEMENTS_URL;

File: src/controllers/update/update.statement.validation.errors.controller.ts [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+
+    const appData = getApplicationData(req.session);
+    const errors = validationResult(req);
+    const inNoChangeJourney = !!appData.update?.no_change;
+
+    if (!errors.isEmpty()) {
+      return renderPage(res, appData, inNoChangeJourney, formatValidationError(errors.array()));
+    }
+
+    if (req.body.statement_resolution === 'statement-resolution-change-information') {
```
Comment (AndreKramer7654) [inline src/controllers/update/update.statement.validation.errors.controller.ts:41]: we have a lot of constants already - should this be an const enum (2 strings)?

File: test/setup.ts [hunk 5]
```diff
@@ -25,4 +25,5 @@ export default () => {
   //  feature flags
   process.env.SHOW_SERVICE_OFFLINE_PAGE = "false";
   process.env.FEATURE_FLAG_ENABLE_SAVE_AND_RESUME_17102022 = "true";
+  process.env.FEATURE_FLAG_ENABLE_UPDATE_STATEMENT_VALIDATION_05072023 = "true";
```
Comment (AndreKramer7654) [inline test/setup.ts:28]: We should default to false. Danger of turning this on somewhere by default.

Comment (conuirchgov) [reply test/setup.ts:28]: I'll just remove the flag from here altogether and mock isActiveFeature 👍

File: src/utils/navigation.ts [hunk 6]
```diff
@@ -379,4 +379,16 @@ export const NAVIGATION: Navigation = {
     previousPage: () => config.UPDATE_ANY_TRUSTS_INVOLVED_URL,
     nextPage: [config.UPDATE_ANY_TRUSTS_INVOLVED_URL]
   },
+  [config.UPDATE_STATEMENT_VALIDATION_ERRORS_URL]: {
+    currentPage: config.UPDATE_STATEMENT_VALIDATION_ERRORS_PAGE,
+    previousPage: (appData?: ApplicationData) => appData?.update?.no_change
```
Comment (AndreKramer7654) [inline src/utils/navigation.ts:384]: An inventive extension of peviousPage which is usually (): string => {}

**Review by AndreKramer7654 — APPROVED** [review]

**Review by corybudd-ch — APPROVED** [review]

**Pull request merged**